### PR TITLE
refactor: contextmenu deprecate clicknotifier methods v23

### LINKED ISSUE #1250: Change ContextMenu to not extend ClickNotifier
`vaadin.version 13.0.6`
If you use `contextMenu.setOpenOnClick(true)`, you can make the context menu open on left mouse button clicks. It would then make sense that using `contextMenu.addClickShortcut` afterwards would enable you to open the context menu using the defined shortcut - currently, using the shortcut does nothing.

### PR DESCRIPTION
**Pull request by ugur-vaadin**
## Description

`ContextMenu` element is not clickable, therefore the implementation of `ClickNotifier` interface is unnecessary. This PR deprecates the use of the methods that are provided by this interface.

This is a temporary change and the interface will be [removed](https://github.com/vaadin/flow-components/pull/4031) in v24.

Fixes #1250 

## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] Refactor

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

### CODE REVIEW

File: vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/GeneratedVaadinContextMenu.java [hunk 1]
```diff
@@ -352,4 +346,34 @@ protected Registration addOpenedChangeListener(
...
+
+    /**
+     * Has no effect for ContextMenu because the element for ContextMenu is not
+     * clickable. Therefore, this method should not be used.
+     *
+     * @see #addClickListener(ComponentEventListener)
+     *
+     * @deprecated since 23.3
+     */
+    @Deprecated(forRemoval = true)
```
Comment (web-padawan) [inline vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/GeneratedVaadinContextMenu.java:358]: Looks like we are not using `forRemoval` elsewhere, so this should be enough:

```suggestion
    @Deprecated()
```

File: vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/GeneratedVaadinContextMenu.java [hunk 2]
```diff
@@ -352,4 +346,34 @@ protected Registration addOpenedChangeListener(
...
+
+    /**
+     * Has no effect for ContextMenu because the element for ContextMenu is not
+     * clickable. Therefore, this method should not be used.
+     *
+     * @see #addClickShortcut(Key, KeyModifier...)
+     *
+     * @deprecated since 23.3
+     */
+    @Deprecated(forRemoval = true)
```
Comment (web-padawan) [inline vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/GeneratedVaadinContextMenu.java:373]:
```suggestion
    @Deprecated()
```

File: vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/GeneratedVaadinContextMenu.java [hunk 3]
```diff
@@ -352,4 +346,34 @@ protected Registration addOpenedChangeListener(
...
+
+    /**
+     * Has no effect for ContextMenu because the element for ContextMenu is not
+     * clickable. Therefore, this method should not be used.
+     *
+     * @see #addClickListener(ComponentEventListener)
+     *
+     * @deprecated since 23.3
+     */
+    @Deprecated()
```
Comment (sissbruecker) [inline vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/GeneratedVaadinContextMenu.java:358]:
```suggestion
    @Deprecated
```
Nit: No need for the parenthesis

File: vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/GeneratedVaadinContextMenu.java [hunk 4]
```diff
@@ -352,4 +346,34 @@ protected Registration addOpenedChangeListener(
...
+
+    /**
+     * Has no effect for ContextMenu because the element for ContextMenu is not
+     * clickable. Therefore, this method should not be used.
+     *
+     * @see #addClickShortcut(Key, KeyModifier...)
+     *
+     * @deprecated since 23.3
+     */
+    @Deprecated()
```
Comment (sissbruecker) [inline vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/GeneratedVaadinContextMenu.java:373]:
```suggestion
    @Deprecated
```
Nit: No need for the parenthesis

**Pull request merged**